tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -70,12 +72,22 @@ object Build {
         linked
       }
 
+      implicit val incCompilationContext: IncCompilationContext =
+        new IncCompilationContext(fconfig.workdir)
+      if (config.compilerConfig.useIncrementalCompilation) {
+        incCompilationContext.collectFromPreviousState()
+      }
+
       // optimize and generate ll
       val generated = {
         val optimized = ScalaNative.optimize(fconfig, linked)
         ScalaNative.codegen(fconfig, optimized)
       }
 
+      if (config.compilerConfig.useIncrementalCompilation) {
+        incCompilationContext.dump()
+      }
```

Comment:
At this point, there is the last usage of incOmpilationContext. We should be able to clear the underlying maps at this point to allow for their GC. Maybe we can introduce method `clear` to the context which would be called now